### ISSUE TITLE
imshow should silence warnings on invalid values, at least when masked

### ISSUE THREAD

**Issue by anntzer** [matplotlib/matplotlib#7746]
mpl 2.0rc2, numpy 1.11.3

An invalid value, whether masked or not, leads to two RuntimeWarnings:
```
imshow(np.ma.masked_invalid([[1, 2], [3, np.nan]]))
```
and
```
imshow([[1, 2], [3, np.nan]])
```
give
```
/usr/lib/python3.5/site-packages/matplotlib/image.py:371: RuntimeWarning: invalid value encountered in less
  rgba[..., 1] = A < 0  # under data
/usr/lib/python3.5/site-packages/matplotlib/image.py:372: RuntimeWarning: invalid value encountered in greater
  rgba[..., 2] = A > 1  # over data
```

I would argue that a warning should be raised in neither case, given that such inputs are supposed to be supported (http://matplotlib.org/examples/pylab_examples/image_masked.html, #6069).

**Comment by efiring** [comment]
You are correct, this is a bug.  I think it is an oversight from the major re-writing of image code.

**Comment by tacaswell** [comment]
Changed my mind on the tagging, this should be fixed before 2.0 final.

**Comment by tacaswell** [comment]
closed by #7780
